src/cls/rgw/cls_rgw_types.cc

```diff
@@ -721,8 +721,7 @@ void cls_rgw_reshard_entry::dump(Formatter *f) const
   encode_json("bucket_id", bucket_id, f);
   encode_json("new_instance_id", new_instance_id, f);
   encode_json("old_num_shards", old_num_shards, f);
-  encode_json("new_num_shards", new_num_shards, f);
-
+  encode_json("tentative_new_num_shards", new_num_shards, f);
```

Comment:
qa/workunits/rgw/test_rgw_reshard.py has test cases based on new_num_shards:
```
Traceback (most recent call last):
  File "/home/ubuntu/cephtest/clone.client.0/qa/workunits/rgw/test_rgw_reshard.py", line 233, in <module>
    main()
  File "/home/ubuntu/cephtest/clone.client.0/qa/workunits/rgw/test_rgw_reshard.py", line 152, in main
    assert json_op[0]['new_num_shards'] == num_shards_expected
KeyError: 'new_num_shards'
```
from http://qa-proxy.ceph.com/teuthology/cbodley-2022-01-28_01:02:31-rgw-wip-cbodley-testing-distro-default-smithi/6644963/teuthology.log

Reply:
we don't run this in upgrade tests, so it should be fine to update tests to use `tentative_new_num_shards` instead

Reply:
Fixed. Thanks, @cbodley .